SonarQube.MSBuild.Tasks/Targets/SonarQube.Integration.targets

```diff
@@ -306,11 +306,9 @@
     <RunCodeAnalysisOnce>true</RunCodeAnalysisOnce>
   </PropertyGroup>
 
-  <!-- We want to override any properties that have been set declaratively in the project -->
-  <Target Name="OverrideCodeAnalysisProperties" Condition=" $(SonarQubeTempPath) != '' "
-          DependsOnTargets="SonarQubeCategoriseProject"
-          BeforeTargets="RunCodeAnalysis" >
-
+  <!-- Check whether or not FxCop should be executed, even if FxCop isn't installed and all targets depending on RunCodeAnalysis are skipped -->
+  <Target Name="ShouldFxCopBeExecuted" Condition=" $(SonarQubeTempPath) != '' "
```

Comment:
The target computes if the ruleset file is present. Perhaps naming it more in the order of "DetectRuleset" or "DetectFxCopRuleset" is more meaningful .